ui/src/app/user.service.ts

```diff
@@ -0,0 +1,39 @@
+import {Injectable} from '@angular/core';
+import {Observable} from 'rxjs'
+import {Currencies, Currency, GraphQLResponse, User} from './app.model';
+import {HttpClient} from '@angular/common/http';
+import {environment} from '../environments/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserService {
+  private httpEndpoint = environment.http_endpoint;
+
+  constructor(private httpClient: HttpClient) {}
+
+  findUser(email: string, what: string[]): Observable<GraphQLResponse<{ findClient: User }>> {
+    const body = {
+      query: 'query FetchName($email: String!){ findClient (email: $email){' + what.join(', ') + '}}',
```

Comment:
I am here to post a meme:

![image](https://user-images.githubusercontent.com/15652452/58431360-a48e5680-80ad-11e9-93ea-070be1991bf2.png)

Reply:
this is actually as smart as we can get with representing everything as strings xd